iam/policy/statement.go

```diff
@@ -166,8 +171,14 @@ func (statement Statement) Equals(st Statement) bool {
 
 // Clone clones Statement structure
 func (statement Statement) Clone() Statement {
-	return NewStatement(statement.SID, statement.Effect, statement.Actions.Clone(),
-		statement.Resources.Clone(), statement.Conditions.Clone())
+	return Statement{
+		SID:        statement.SID,
+		Effect:     statement.Effect,
+		Actions:    statement.Actions,
+		NotActions: statement.NotActions,
+		Resources:  statement.Resources,
+		Conditions: statement.Conditions,
+	}
```

Comment:
why is this being changed? @eirture

Reply:
This method is a clone operation. A new field `NoActions` add to the statement, when we clone a statement we should copy the `NotActions` too. Is this right?

Reply:
yes but you need to enhance the NewStatement() to take `NotActions.Clone()` as input what is happening right now is not a copy AFAIK.

Reply:
yes but I think the NewStatement function is a promise to users, we can't change parameters anymore. Can we add a new function to create a statement? Could you give me some advice?

Reply:
yes you can of course add, the only consumer of this function is MinIO - we don't expect users to use this package outside of MinIO. That is out scope for us.

Reply:
you can also add NewStatementWithNotActions() if you want to keep previous function as is.